tsl/test/expected/bgw_custom.out

```diff
@@ -269,3 +269,121 @@ total_successes        | 0
+    r RECORD;
+BEGIN
+    FOR i in 1..spins
+    LOOP
+    SELECT total_successes, total_failures FROM _timescaledb_internal.bgw_job_stat WHERE job_id=job_param_id INTO r;
+    IF (r.total_failures > 0) THEN
+        EXIT;
+    ELSEIF (r.total_successes = expected_runs) THEN
+        RETURN true;
+    ELSEIF (r.total_successes > expected_runs) THEN
+        RAISE 'num_runs > expected';
+    ELSE
+        PERFORM pg_sleep(0.1);
+    END IF;
+    END LOOP;
+    RETURN false;
+END
+$BODY$;
+TRUNCATE custom_log;
+-- Nested procedure call
+CREATE OR REPLACE PROCEDURE custom_proc_nested(job_id int, args jsonb) LANGUAGE PLPGSQL AS
+$$
+BEGIN
+  INSERT INTO custom_log VALUES($1, $2, 'custom_proc_nested 1 COMMIT');
+  COMMIT;
+  INSERT INTO custom_log VALUES($1, $2, 'custom_proc_nested 2 ROLLBACK');
+  ROLLBACK;
+  INSERT INTO custom_log VALUES($1, $2, 'custom_proc_nested 3 COMMIT');
+  COMMIT;
+END
+$$;
+CREATE OR REPLACE PROCEDURE custom_proc3(job_id int, args jsonb) LANGUAGE PLPGSQL AS
+$$
+BEGIN
+    CALL custom_proc_nested(job_id, args);
+END
+$$;
+CREATE OR REPLACE PROCEDURE custom_proc4(job_id int, args jsonb) LANGUAGE PLPGSQL AS
+$$
+BEGIN
+    INSERT INTO custom_log VALUES($1, $2, 'custom_proc4 1 COMMIT');
+    COMMIT;
+    INSERT INTO custom_log VALUES($1, $2, 'custom_proc4 2 ROLLBACK');
+    ROLLBACK;
+    RAISE EXCEPTION 'forced exception';
+    INSERT INTO custom_log VALUES($1, $2, 'custom_proc4 3 ABORT');
+    COMMIT;
+END
+$$;
+-- Remove any default jobs, e.g., telemetry
+\c :TEST_DBNAME :ROLE_SUPERUSER
+TRUNCATE _timescaledb_config.bgw_job RESTART IDENTITY CASCADE;
+NOTICE:  truncate cascades to table "bgw_job_stat"
+NOTICE:  truncate cascades to table "bgw_policy_chunk_stats"
+\c :TEST_DBNAME :ROLE_DEFAULT_PERM_USER
+SELECT add_job('custom_proc2', '1h', config := '{"type":"procedure"}'::jsonb, initial_start := now()) AS job_id_1 \gset
+SELECT add_job('custom_proc3', '1h', config := '{"type":"procedure"}'::jsonb, initial_start := now()) AS job_id_2 \gset
+\c :TEST_DBNAME :ROLE_SUPERUSER
+-- Start Background Workers
+SELECT _timescaledb_internal.start_background_workers();
+ start_background_workers 
+--------------------------
+ t
+(1 row)
+
+-- Wait for jobs
+SELECT wait_for_job_to_run(:job_id_1, 1);
+ wait_for_job_to_run 
+---------------------
+ t
+(1 row)
+
+SELECT wait_for_job_to_run(:job_id_2, 1);
+ wait_for_job_to_run 
+---------------------
+ t
+(1 row)
+
+-- Check results
+SELECT * FROM custom_log ORDER BY job_id, extra;
+ job_id |         args          |            extra            |      runner       
+--------+-----------------------+-----------------------------+-------------------
+   1000 | {"type": "procedure"} | custom_proc 1 COMMIT        | default_perm_user
+   1000 | {"type": "procedure"} | custom_proc 3 COMMIT        | default_perm_user
+   1001 | {"type": "procedure"} | custom_proc_nested 1 COMMIT | default_perm_user
+   1001 | {"type": "procedure"} | custom_proc_nested 3 COMMIT | default_perm_user
+(4 rows)
+
```

Comment:
If we delete jobs 1 and 2 and then truncate custom_log, the output will be cleaner for the next job run.
select delete_job(:job_id_1);

Reply:
Fixed!